lib/public/components/common/dataExport/exportTriggerAndModal.js

```diff
@@ -0,0 +1,147 @@
+    }, [
+        ...fields
+            .filter((name) => !['id', 'actions'].includes(name))
+            .map((name) => h('option', {
+                value: name,
+                selected: selectedFields.length ? selectedFields.includes(name) : false,
+            }, name)),
+    ]);
+
+    const exportTypeSelectionLabels = [
+        h('label.form-check-label.f4.mt1', 'Export type'),
+        h('label.form-check-label.f6', 'Select output format'),
+    ];
+
+    const exportTypeSelect = h('.flex-row.g3', exportTypes.map((exportType) => {
+        const id = `data-export-type-${exportType}`;
+        return h('.form-check', [
+            h('input.form-check-input', {
+                id,
+                type: 'radio',
+                value: exportType,
+                checked: selectedExportType.length ? selectedExportType.includes(exportType) : false,
+                name: 'export-type',
+                onclick: () => exportModel.setSelectedExportType(exportType),
+            }),
+            h('label.form-check-label', {
+                for: id,
+            }, exportType),
+        ]);
+    }));
+
+    const dataAvailable = exportModel.items.match({ Success: () => true, Other: () => false });
+
+    const exportBtn = h('button.shadow-level1.btn.btn-success.mt2#send', {
+        disabled: !(fieldsSelected && dataAvailable),
+        onclick: async () => {
+            await exportModel.createExport();
+            modalHandler.dismiss();
+        },
+    }, dataAvailable ? 'Export' : 'Loading data...');
+
+    const dataLength = exportModel.items.match({ Success: ({ length } = {}) => length, Other: () => null });
+    const { totalExistingItemsCount } = exportModel;
+
+    const truncatedDataInfo = dataLength && dataLength < totalExistingItemsCount
+        ? h(
+            '#truncated-export-warning.warning',
+            `The data export is limited to ${dataLength} entries, only the most recent data will be exported`,
+        )
+        : null;
+
+    return [
+        truncatedDataInfo,
+        fieldsSelectionLabels,
+        fieldsSelection,
+        exportTypeSelectionLabels,
+        exportTypeSelect,
+        exportBtn,
+    ];
+};
+
+const errorDisplay = () => h('.danger', 'Data fetching failed');
+
+/**
+ * A function to construct the exports data screen
+ *
+ * @param {DataExportModel} exportModel export model
+ * @param {ModalHandler} modalHandler The modal handler, used to dismiss modal after export
+ * @return {Component} Return the view of the inputs
+ */
+const exportModal = (exportModel, modalHandler) => {
+    exportModel.callForData();
+    const dataLoading = exportModel.items.match({ Loading: () => true, Other: () => false });
+
+    return h('div#export-data-modal', [
+        h('.flex-row', [
+            h('h2.w-50', 'Export data'),
+            dataLoading ? h('.w-50', spinner({ size: 2, absolute: false })) : null,
+        ]),
+        exportModel.items.match({
```

Comment:
if you already check the remote data state of `items` in the return, why not add a `Loading` state there instead of extracting the dataLoading ?